pkg/provider/provider.go

```diff
@@ -101,23 +112,57 @@ func DSN(s *schema.ResourceData) (string, error) {
 		region = ""
 	}
 
-	dsn, err := gosnowflake.DSN(&gosnowflake.Config{
-		Account:  account,
-		User:     username,
-		Region:   region,
-		Password: password,
-		Role:     role,
-	})
-
-	if browserAuth {
-		dsn, err = gosnowflake.DSN(&gosnowflake.Config{
-			Account:       account,
-			User:          username,
-			Region:        region,
-			Role:          role,
-			Authenticator: gosnowflake.AuthTypeExternalBrowser,
-		})
+	config := gosnowflake.Config{
+		Account: account,
+		User:    user,
+		Region:  region,
+		Role:    role,
 	}
 
-	return dsn, err
+	if len(pathPrivateKey) != 0 {
```

Comment:
Maybe I am missing something, but shouldn't this be `!= nil`?

Reply:
I thought so too, but when I tried your suggestion I got this:
```
# github.com/chanzuckerberg/terraform-provider-snowflake/pkg/provider
pkg/provider/provider.go:122:20: invalid operation: privateKeyPath != nil (mismatched types string and nil)
```